src/taipy/gui_core/GuiCoreLib.py

```diff
@@ -690,6 +689,23 @@ def get_data_node_tabular_columns(self, datanode: DataNode, id: str):
                 return None
         return None
 
+    def get_data_node_chart_config(self, datanode: DataNode, id: str):
+        if (
+            id
+            and isinstance(datanode, DataNode)
+            and id == datanode.id
+            and (dn := core_get(id))
+            and isinstance(dn, DataNode)
+            and dn.is_ready_for_reading
```

Comment:
Is there a situation where core_get(datanode.id) would be different from datanode?
I guess yes if it was deleted...